integration_tests/test_str_01.py

```diff
@@ -45,6 +45,14 @@ def test_str_repeat():
     assert 3*a*3 == "XyzXyzXyzXyzXyzXyzXyzXyzXyz"
     assert a*-1 == ""
 
+def test_str_join():
+    a: str
+    a = ","
+    p:list[str] = ["a","b"]
+    res:str = a.join(p)
+    assert res == "a,b"
```

Comment:
Very nice. Let's also add a test where "a" is more than one character, say: `a = ", "` and `assert rest == "a, b"`.

Reply:
Actually it doesn't work for `",".join(["a,""b"])`.
 I need to make some changes, I have added a PR as a part of pycon India, I will continue after coming back to my hometown.

Reply:
I see. We can still merge it as a start, then we can improve it.

Reply:
sure, I have added a new test.

Reply:
I see it, thanks.